gssapi/tests/test_raw.py

```diff
@@ -705,7 +707,12 @@ def test_rfc5587(self):
                 display_out.short_desc.should_be_a(bytes)
                 display_out.long_desc.should_be_a(bytes)
 
-        attrs = set([last_attr])
+                if mech_attr not in attrs_dict:
+                    attrs_dict[mech_attr] = 0
+                attrs_dict[mech_attr] += 1
```

Comment:
the comment above needs to reflect the commit message comments to explain what's going on here.
Also, don't you need to ensure that last_mech actually has the attr -- maybe we should record the mechs that have each attr?